### PR TITLE
Insert new scaling algorithm

### PR DESCRIPTION
**Pull request by skjerns**
as suggested in #29

### CODE REVIEW

File: visbrain/io/read_sleep.py [hunk 1]
```diff
@@ -195,10 +196,14 @@ def __init__(self, data, channels, sf, hypno, href, preload, use_mne,
...
-        if np.abs(np.ptp(data, 0).mean()) < 0.1:
-            warn("Wrong data amplitude for Sleep software.")
-            data *= 1e6
+        # Assume that the inter-quartile amplitude of EEG data is ~50 uV
+        iqr_data = iqr(data, axis=1)
+
+        for idx_chan, iqr_chan in enumerate(iqr_data):
+            if iqr_chan < 1:
+                mult_fact = np.floor(np.log10(50 / iqr_chan))
+                warn("Wrong channel data amplitude. Multiplying data amplitude by 10^%i" % mult_fact)
```
Comment (EtienneCmb) [inline visbrain/io/read_sleep.py]: - Line too long. See : https://travis-ci.org/EtienneCmb/visbrain/jobs/444620569
- `mult_fact` is a float, not int so you can't use `%i`

**Comment by EtienneCmb** [comment]
> Thanks @skjerns for the PR and @EtienneCmb for your input! I think a final version could be:
> 
> ```
> from scipy.stats import iqr
> 
> # Assume that the inter-quartile amplitude of EEG data is ~50 uV
> iqr_chan = iqr(data[:, :int(data.shape[1] / 4)], axis=-1)
> bad_iqr = iqr_chan < 1.
> 
> if np.any(bad_iqr):
>     mult_fact = np.zeros_like(iqr_chan)
>     mult_fact[bad_iqr] = np.floor(np.log10(50 / iqr_chan[bad_iqr]))
>     data *= 10 ** mult_fact[..., np.newaxis]
> ```
> Advantages:
> 
> * no for loop
> * we only compute the IQR on 1/4 of the data (should be enough to have a good idea of what the real IQR is, and speed up the computation by a factor of four)
> 
> I did some tests on my computer and for ~8 hours of data, sampled at 100 Hz and with 20 channels, the computation of the IQR takes ~300 ms (and 1.3 second for the full data).
> 
> Do you see any way that we could further speed up / improve the code?

The 300ms comes from the `iqr()` function?

**Comment by raphaelvallat** [comment]
Yes. If we want to remove the call to scipy, the `iqr` function is essentially doing:
` np.diff(np.percentile(x, [25, 75], axis=1), axis=0)[0]`

The numpy version being just a bit faster (but this is almost negligible).

**Comment by EtienneCmb** [comment]
> The numpy version being just a bit faster (but this is almost negligible).

Probably because there's checks inside the `iqr` function. If it's only a question os ms, I propose to keep the `iqr` version.

**Comment by EtienneCmb** [comment]
@raphaelvallat can you please take a moment to review this PR, we could merge it today because tests passed

**Pull request merged**